src/qinst/network_inst.py

```diff
@@ -0,0 +1,81 @@
+import ipaddress
+import socket
+import time
+
+from qinst.instrument import Instrument
+
+
+class NetworkInst(Instrument):
+    """Base class for instruments communicating via network connection."""
+
+    def __init__(
+        self,
+        name: str,
+        address: str,
+        port: int = 5025,  # Keysight instruments standard port
+        timeout: int = 10,
+        sleep: float = 0.1,
+        no_delay=True,
+    ):
+        super().__init__(name, address)
+
+        # Validate IP
+        _ = ipaddress.ip_address(address)
+        self.port = port
+        self.sleep = sleep
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.timeout = timeout
+        self.no_delay = no_delay
+
+    def __del__(self):
+        self.disconnect()
```

Comment:
Da altre parti abbiamo scritto cose come:
```suggestion
    def __del__(self):
    	is self.serial.is_open():
            self.disconnect()
```
Magari dovremmo fare qualcosa di simile anche qui (forse lo hai già fatto non capisco totalmente)